src/ansys/pytwin/evaluate/tbrom.py

```diff
@@ -171,12 +164,81 @@ def _reduce_field_input(self, name: str, snapshot: Union[str, Path, np.ndarray])
                 self._infmcs[name][item] = mc[index]
                 index = index + 1
 
+    def project_on_mesh(self, mesh: pv.DataSet, interpolate: bool, named_selection: str = None):
```

Comment:
thanks Ed. The code has been adapted a bit differently : 
- explicit import of tqdm to define _HAS_TQDM to True/False
- optional argument of progress_bar for method project_on_mesh
- propagation of progress_bar to the call of interpolate, but in case progress_bar is False and _HAS_TQDM is True, progress_bar is automatically set to True